proposals/0291-allows-navigation-apps-to-access-information-about-Wi-Fi-networks.md

```diff
@@ -178,12 +239,117 @@ No alternatives were identified.
                 return;
             } else {
                 Log.d(TAG, "onTransportDisconnected - " + disconnectedTransport.getType().name());
-+               if (disconnectedTransport.getType() == TransportType.TCP && secondaryTransportParams != null){
-+                   if (activeTransports.containsValue(disconnectedTransport)) {
-+                       //If the established TCP connection is disconnected, the corresponding IP and port are invalid and should be removed from the list.
-+                       // Otherwise, istransportforserviceavailable is always true after disconnection
++               if (disconnectedTransport.getType() == TransportType.TCP && secondaryTransportParams != null) {
++                   if (activeTransports.containsValue(disconnectedTransport)
++                           && (transportManager != null && (transportManager.isWifiConnected() || transportManager.isWifiAPStateEnabled()))) {
++                       // If the established TCP connection is disconnected, the corresponding IP and port are invalid and should be removed from the list.
++                       // Otherwise, isTransportForServiceAvailable is always true after disconnection
++                       // Do not remove when Wifi is connected or AP is enabled in HS side, because app need use it when connected again.

```

Comment:
```suggestion
+                       // Do not remove when WiFi is connected or access point is enabled on mobile device, because the app needs to use it when connected again.
```